tests/platform_test_harness/graphics_platform_test_harness.cpp

```diff
@@ -0,0 +1,676 @@
+
+        orange_displacement = bounce_within(
+            orange_rectangle->screen_position(),
+            orange_displacement,
+            delta_t,
+            bounding_box);
+        orange_rectangle->set_position(
+            orange_rectangle->screen_position().top_left +
+            delta_t_ms * orange_displacement);
+
+        aubergine_displacement = bounce_within(
+            aubergine_rectangle->screen_position(),
+            aubergine_displacement,
+            delta_t,
+            bounding_box);
+        aubergine_rectangle->set_position(
+            aubergine_rectangle->screen_position().top_left +
+            delta_t_ms * aubergine_displacement);
+
+        for(auto const& renderer : renderers)
+        {
+            renderer->render({orange_rectangle, aubergine_rectangle});
+        }
+    }
+}
+
+auto test_platform_supports_accelerated_wayland_clients(mg::GraphicBufferAllocator const& allocator) -> bool
+{
+    if (dynamic_cast<mg::WaylandAllocator const*>(&allocator))
+    {
+        std::cout << "Platform supports Wayland EGL" << std::endl;
+        return true;
+    }
+    else
+    {
+        std::cout << "Platform does *not* support Wayland EGL" << std::endl;
+        return false;
+    }
+}
+}
+
+int main(int argc, char const** argv)
+{
+    if (argc != 2)
+    {
+        std::cerr << "Usage: " << argv[0] << " PLATFORM_DSO" << std::endl;
+        return -1;
+    }
+
+    mir::SharedLibrary platform_dso{argv[1]};
+
+    MinimalServerEnvironment config;
+
+    bool success = true;
+    success &= test_probe(platform_dso, config);
+    if (success)
+    {
+        if (auto display_platform = test_display_platform_construction(platform_dso, config))
+        {
+            if(auto display = test_display_construction(*display_platform, config))
+            {
+                success &= test_display_supports_gl(*display);
+                success &= dump_egl_config(*display);
+                success &= test_display_has_at_least_one_enabled_output(*display);
+                success &= test_display_buffers_support_gl(*display);
+                basic_display_swapping(*display);
+
+                if (auto render_platform = test_render_platform_construction(platform_dso, config))
+                {
+                    auto buffer_allocator = render_platform->create_buffer_allocator(*display);
+                    success &= test_platform_supports_accelerated_wayland_clients(*buffer_allocator);
+
+                    basic_software_buffer_drawing(
+                    *display,
+                    *buffer_allocator,
+                    *config.render_factory());
+                }
+                else
+                {
+                    success = false;
+                }
+            }
+            else
+            {
+                success = false;
+            }
+        }
+        else
+        {
+            success = false;
+        }
+    }
+    return success ? 0 : -1;
```

Comment:
For the sake of clarity when running from the command-line let's `puts(success ? "SUCCESS", "FAILURE")`

Reply:
Done

Reply:
I think you misread? I just wanted something shown on the console to indicate the result.

Reply:
Hah! I did indeed.

I forgot that not all users have terminals that visually differentiate between “last command returned success” and “last command returned failure”.